apps/examples/testcase/le_tc/compression/tc_compress_read.c

```diff
@@ -22,46 +22,312 @@
+ *
+ * Description:
+ *   Testing compression and decompression of a file when the
+ *   output size after decompression is assigned same as input size
+ *   before decomrpession . 
+ *   This can be taken as the case where output size is kept less than
+ *   actual size of decompressed data.
+ *
+ * Returned Value:
+ *   None
+ ****************************************************************************/
+static void tc_comp_decomp_buffer_lesser_output_size(void)
+{
+	int ret_chk;
+	fd = get_compfd();
+
+	/*If FAIL : Failed to open testcase driver*/
+	TC_ASSERT_GEQ("get_compfd", fd, OK);
+	
+	// initializing comp_tc_data
+	ret_chk = initialize_tc_data();
+	TC_ASSERT_EQ_CLEANUP("compression data initialize", ret_chk, OK, uninitialize_tc_data());
+	
+	// Test compression method 
+	ret_chk = ioctl(fd, COMPIOC_COMPRESS, comp_tc_data);
+	TC_ASSERT_EQ_CLEANUP("compress block", ret_chk, OK, uninitialize_tc_data());
+	
+	decomp_tc_data->input_size = comp_tc_data->output_size;
+	decomp_tc_data->output_size = decomp_tc_data->input_size;
+	decomp_tc_data->output_buffer = (unsigned char *)malloc(decomp_tc_data->output_size);
+	
+	// Test decompression method
+	ret_chk = ioctl(fd, COMPIOC_DECOMPRESS, decomp_tc_data);
+	if (CONFIG_COMPRESSION_TYPE == MINIZ_TYPE) {
+		TC_ASSERT_EQ_CLEANUP("decompress block", ret_chk, ENOMEM, uninitialize_tc_data());
+	}
+	if (CONFIG_COMPRESSION_TYPE == LZMA_TYPE) {
+		TC_ASSERT_EQ_CLEANUP("decompress block", ret_chk, ENOMEM, uninitialize_tc_data());
+		TC_ASSERT_LEQ_CLEANUP("input size after decompress", decomp_tc_data->input_size, comp_tc_data->output_size - LZMA_PROPS_SIZE, uninitialize_tc_data());
+		TC_ASSERT_EQ_CLEANUP("output size after decompress", decomp_tc_data->output_size, comp_tc_data->output_size, uninitialize_tc_data());
+
+		//check the data before compression and after decompression
+		for (int i = 0; i < comp_tc_data->output_size; i++){
+			TC_ASSERT_EQ_CLEANUP("output after decompress", comp_tc_data->input_buffer[i], decomp_tc_data->output_buffer[i], uninitialize_tc_data());
+		}
+	}
+	
+	uninitialize_tc_data();
+	TC_SUCCESS_RESULT();
+}
 
+/****************************************************************************
+ * Name: tc_comp_decomp_buffer_greater_output_size
+ *
+ * Description:
+ *   Testing compression and decompression of a file when the
+ *   output size after decompression is very large than input size
+ *   before decomrpession . 
+ *   This can be taken as the case where output size is larger than
+ *   actual size of decompressed data.
+ *
+ * Returned Value:
+ *   None
+ ****************************************************************************/
+static void tc_comp_decomp_buffer_greater_output_size(void)
+{
+	int ret_chk;
+	fd = get_compfd();
+
+	/*If FAIL : Failed to open testcase driver*/
+	TC_ASSERT_GEQ("get_compfd", fd, OK);
+	
+	// initializing comp_tc_data
+	ret_chk = initialize_tc_data();
+	TC_ASSERT_EQ_CLEANUP("compression data initialize", ret_chk, OK, uninitialize_tc_data());
```

Comment:
In there is a failure in init, then no memory is allocated. So, we dont need to call uninit.